### ISSUE TITLE
Tabs - Left Scroll

### ISSUE THREAD

**Issue by zanderson9** [vincenzopalazzo/material-ui-swing#146]
Hi Vincent,

is it possible to add the following improvement to the tabs scrolling effect, as per JMARS user comment:

"When tabs are hidden past the **right** edge of the dialog, there is a **separator line** to tell user that you are not seeing everything.  If you use the arrow to scroll to the **right**, the left-most tabs disappear, but there is no **separator line** on the left end to let user know there are missing tabs. This should be fixed."

Can we add same separator line when scrolling left?

<img width="786" alt="scroll-right" src="https://user-images.githubusercontent.com/49419742/88728389-7f004100-d0e6-11ea-97dd-eb154391e048.png">

<img width="809" alt="scroll-left" src="https://user-images.githubusercontent.com/49419742/88728407-8889a900-d0e6-11ea-819f-a6f1be42fa24.png">

**Comment by vincenzopalazzo** [comment]
Hi @zanderson9,

Unfortunately, the line is not inside the material-ui-swing style, is only swing refuse. unfortunately at the moment, this change is not possible also because by material style this feature shouldn't exist.

The user can see that there is more tab to see from the button disabled, like in your example.

Maybe in the future but, I'm not sure about this style

**Comment by zanderson9** [comment]
thank you for a prompt reply.

Then how about if we remove the separator line altogether?

Can we have the scrolling effect with just **left** and **right** arrows, like for example in Gmail desktop UI?

<img width="504" alt="Tabs" src="https://user-images.githubusercontent.com/49419742/88730391-ce487080-d0ea-11ea-8368-90531d3d1597.png">

**Comment by vincenzopalazzo** [comment]
Hi @zanderson9,

The line does not exist inside the material-ui-swing style, because it has the same color as the background. This effect is a custom effect inside JMars. You can add the following propriety

```java
UIManager.put("TabbedPane.font", ThemeFont.getBold().deriveFont(FONTS.ROBOTO_TAB.fontSize()));
        UIManager.put("TabbedPane.selected", this.getSelected());
        UIManager.put("TabbedPane.highlight", this.getUnderlinehighlight()); //line under tabs
        UIManager.put("TabbedPane.background", this.getBackground());
        UIManager.put("TabbedPane.foreground", this.getForeground());
        UIManager.put("TabbedPane.borderHighlightColor", this.getHighlight());
        UIManager.put("TabbedPane.contentAreaColor", this.getContentareacolor());
       // UIManager.put("TabbedPane.disabledForeground", this.getDisableTextColor());
        UIManager.put("TabbedPane[focus].colorLine", this.getFocuscolorline());        
        UIManager.put("TabbedPane.selectionForeground", this.getFocuscolorline());
        UIManager.put("TabbedPane.tabInsets",  new InsetsUIResource(0, getTabspace(), 0, getTabspace()));
        //UIManager.put("TabbedPane.selectedTabPadInsets", new InsetsUIResource(0, 10, 0, 10));
        UIManager.put("TabbedPane.shadow", this.getBackground());
        UIManager.put("TabbedPane.darkShadow", this.getBackground());
        UIManager.put("TabbedPane.light", this.getBackground());
        //TODO change the color order in the version official 1.1.1, this is a bug
        UIManager.put("TabbedPane[scrollButton].iconLeft", MaterialImageFactory.getInstance().getImage(
                MaterialIconFont.CHEVRON_LEFT,
                this.getDisableTextColor()
        ));
        UIManager.put("TabbedPane[scrollButton].disabledIconLeft", MaterialImageFactory.getInstance().getImage(
                MaterialIconFont.CHEVRON_LEFT,
                this.getForeground()
        ));

        UIManager.put("TabbedPane[scrollButton].iconRight", MaterialImageFactory.getInstance().getImage(
                MaterialIconFont.CHEVRON_RIGHT,
               this.getForeground()
        ));
        UIManager.put("TabbedPane[scrollButton].disabledIconRight", MaterialImageFactory.getInstance().getImage(
                MaterialIconFont.CHEVRON_RIGHT,
                this.getDisableTextColor()
        ));
```

or you can found the difference between the propriety, I don't remember what is the propriety, sorry :) 
I work this MaterialTheming and I forgot the name of proprity.

I hope this answer can help.

**Comment by zanderson9** [comment]
Hi Vincent,

I figured out which properties to use to make the  separator "cropped" line invisible.

These are

UIManager.put("TabbedPane.darkShadow", ... )  And
UIManager.put("TabbedPane.shadow", ...)

If to set them to the same color as background, then they are invisible. 

At some point, I removed those properties from my settings. That is why a default line color was used and separator was visible. Now I restored those 2 settings and can control the separator line appearance.

the actual drawing of those lines happens in BasicTabbedPaneUI, not in your library :-)

Thank you.
